infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -0,0 +1,198 @@
+import org.wso2.carbon.testgrid.common.Script;
+import org.wso2.carbon.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.carbon.testgrid.common.util.StringUtil;
+import org.wso2.carbon.testgrid.utils.EnvVariableUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is responsible for creating the AWS infrastructure.
+ *
+ * @since 1.0.0
+ */
+public class AWSManager {
+
+    private Infrastructure infra;
+    private static final Log log = LogFactory.getLog(AWSManager.class);
+
+    /**
+     * This constructor creates AWS deployer object and validate AWS related environment variables are present.
+     *
+     * @param awsKeyVariableName    Environment variable name for AWS ACCESS KEY.
+     * @param awsSecretVariableName Environment variable name for AWS SECRET KEY.
+     * @throws TestGridInfrastructureException Throws exception when environment variables are not set.
+     */
+    public AWSManager(String awsKeyVariableName, String awsSecretVariableName) throws TestGridInfrastructureException {
+        String awsIdentity = EnvVariableUtil.readEnvironmentVariable(awsKeyVariableName);
+        String awsSecret = EnvVariableUtil.readEnvironmentVariable(awsSecretVariableName);
+        if (StringUtil.isStringNullOrEmpty(awsIdentity) || StringUtil.isStringNullOrEmpty(awsSecret)) {
+            throw new TestGridInfrastructureException("AWS Credentials must be set as environment variables");
+        }
+    }
+
+    /**
+     * This method initiates creating infrastructure through CLoudFormation.
+     *
+     * @param script       Script object containing the CF details.
+     * @param infraRepoDir Path of TestGrid home location in file system as a String.
+     * @return a Deployment object with created infrastructure details.
+     * when an error occurs while reading the cf template file.
+     * @throws TestGridInfrastructureException AWS related error caused by malformed CF template.
+     */
+    public Deployment createInfrastructure(Script script, String infraRepoDir)
+            throws TestGridInfrastructureException {
+        String cloudFormationName = script.getName();
+        AmazonCloudFormation stackbuilder = AmazonCloudFormationClientBuilder.standard()
+                .withCredentials(new EnvironmentVariableCredentialsProvider())
+                .withRegion(infra.getRegion())
+                .build();
+
+        CreateStackRequest stackRequest = new CreateStackRequest();
+        stackRequest.setStackName(cloudFormationName);
+        try {
+            String file = new String(Files.readAllBytes(Paths.get(infraRepoDir, this.infra.getName(),
+                    "AWS", "Scripts", script.getFilePath())));
+            stackRequest.setTemplateBody(file);
+            log.info("Created a CloudFormation Stack with the name :" + stackRequest.getStackName());
```

Comment:
You can't say "Created a CloudFormation Stack" in the log here because the create call is not yet invoked. Better to include debug logs with traceable information.